Locker/Locker/Locker.swift

```diff
@@ -0,0 +1,246 @@
+//
+//  Locker.swift
+//  Locker
+//
+//  Created by Zvonimir Medak on 19.10.2021..
+//  Copyright © 2021 Infinum. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+@objcMembers
+public class Locker: NSObject {
+
+    // MARK: - Public properties
+
+    public static var userDefaults: UserDefaults? {
+        get {
+            return currentUserDefaults == nil ? UserDefaults.standard : currentUserDefaults
+        }
+        set {
+            currentUserDefaults = currentUserDefaults == nil ? UserDefaults.standard : newValue
+        }
+    }
+
+    public static var biometricsSettingsDidChange: Bool {
+        LockerHelpers.biometricsSettingsChanged
+    }
+
+    public static var isRunningFromTheSimulator: Bool {
+    #if targetEnvironment(simulator)
+        return true
+    #else
+        return false
+    #endif
+    }
+
+    public static var deviceSupportsAuthenticationWithBiometrics: BiometricsType {
+        LockerHelpers.deviceSupportsAuthenticationWithBiometrics
+    }
+
+    public static var configuredBiometricsAuthentication: BiometricsType {
+        LockerHelpers.configuredBiometricsAuthentication
+    }
+
+    // MARK: - Private properties
+
+    private static var currentUserDefaults: UserDefaults?
+
+    // MARK: - Handle secrets (store, delete, fetch)
+
+    @objc(setSecret:forUniqueIdentifier:error:)
+    public static func setSecret(_ secret: String, for uniqueIdentifier: String) throws {
```

Comment:
```suggestion
    public static func setSecret(_ secret: String, for uniqueIdentifier: String, completion: ((LockerError?) -> Void)? = nil) {
```
The main reason why this approach is better is that `setSecret` is an async function. If someone wants to know when storing is finished, it could implement a completion block and wait for its result. If in completion we get an error, then storing is finished with error, if error nil, then storing is finished successfully :) 

So in short, if we just throw, we will lose info when storing, in general, is finished. And by adding the optional completion block, we can give to the user the possibility to choose if he/she wants to get the completion info or not.